src/Microsoft.AspNetCore.Dispatcher/RouteCreationException.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Dispatcher
+{
+    /// <summary>
+    /// The exception that is thrown for invalid routes or constraints.
+    /// </summary>
+    public class RouteCreationException : Exception
```

Comment:
We don't need this type

Reply:
What type of exception should be thrown when a constraint cannot be created?